modules/mono/glue/GodotSharp/Godot.SourceGenerators.Internal/UnmanagedCallbacksGenerator.cs

```diff
@@ -168,9 +168,7 @@ private void GenerateInteropMethodImplementations(GeneratorExecutionContext cont
             {
                 var parameter = callback.Parameters[i];
 
-                source.Append(parameter.ToDisplayString());
-                source.Append(' ');
-                source.Append(parameter.Name);
+                source.Append(parameter.GetSyntax().WithAttributeLists(default));
```

Comment:
I don't like that we are relying on `ToString` here. Would it be better to be more explicit instead?

```csharp
var parameter = callback.Parameters[i];

AppendRefKind(source, parameter.RefKind);
source.Append(' ');
source.Append(parameter.Type.FullQualifiedNameIncludeGlobal());
source.Append(' ');
source.Append(parameter.Name);
```

Reply:
It's not just *a* ToString, but this pretty explicitly returns the source for that node (without leading and trailing white space / comments), but mostly his is basically taken from my net7 PR where this used because there is no good way to get the presence of the scoped modifier.

This basically takes the source code from the original declaration and copies it into the output, this should be more future proof than the suggestion, but i am not super attached to it if you think just building the parameter manually is better.

Reply:
(Unrelated to this PR, but `ScopedKind` is if the parameter is a scoped reference and does not directly correlate to if this parameter needs to have the `scoped` modifier to be compatible with the original declaration, so that makes codedgen use a bit of a pain)

Reply:
I don't think so, take a look at this code:

https://github.com/dotnet/roslyn/blob/5e0518c3897ec3f59d6199b280b8a3e77ff7abe3/src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.cs#L813-L849

It adds the `scoped` keyword based on the value of the parameter's `ScopedKind` property (and a few other properties). The `ScopedKind` property alone is not enough because the `this` and `out` parameters are implicitly scoped, but other than that it seems to be what we'd want.

Alternatively, we could retrieve the declaration syntax of the parameter and look for `scoped` in the modifiers of the syntax node; but this seems more brittle.

Reply:
Switched to your suggestion, because mine doen't emit fully qualified type names, and the using in the generated and orignal file are different.

Left out the `scoped` bits for now, we can add them later when we switch to langversion 11+, which has that modifier.